e2e/stack_monitoring_test.go

```diff
@@ -0,0 +1,614 @@
+	allowedDeletionsExtra  []string
+	allowedInsertionsExtra []string
+	handleSpecialCases     func(docType string, legacy *gabs.Container, metricbeat *gabs.Container) error
+}
+
+// @collectionMethod1 the collection method to be used. Valid values: legacy, metricbeat
+// @collectionMethod2 the collection method to be used. Valid values: legacy, metricbeat
+func (sm *StackMonitoringTestSuite) checkDocumentsStructure(
+	collectionMethod1 string, collectionMethod2 string) error {
+
+	log.Debugf("Compare the structure of the %s documents with the structure of the %s documents", collectionMethod1, collectionMethod2)
+
+	return assertHitsEqualStructure(sm, collectionMethod1, collectionMethod2)
+}
+
+// checkProduct sets product name, in lowercase, setting the product port based on a validation:
+// if the product is not supported, then an error is thrown
+func (sm *StackMonitoringTestSuite) checkProduct(product string, collectionMethod string) error {
+	sm.Product = strings.ToLower(product)
+
+	env := map[string]string{
+		"monitoringEsHost": "monitoringEs", // monitoring elasticsearch service name
+		"monitoringEsPort": "9200",         // monitoring elasticsearch port
+	}
+
+	log.Debugf("Enabling %s collection, sending %s metrics to the monitoring instance", collectionMethod, sm.Product)
+
+	productIndexID := sm.Product
+
+	switch {
+	case sm.Product == "elasticsearch":
+		sm.Port = strconv.Itoa(9201)
+		sm.handleSpecialCases = func(docType string, legacy *gabs.Container, metricbeat *gabs.Container) error {
+			return fmt.Errorf("Not supported yet")
+		}
+	case sm.Product == "kibana":
+		sm.Port = strconv.Itoa(5602)
+		sm.allowedDeletionsExtra = []string{
+			"kibana_stats.response_times.max",
+			"kibana_stats.response_times.average",
+		}
+
+		sm.handleSpecialCases = func(docType string, legacy *gabs.Container, metricbeat *gabs.Container) error {
+			// Internal collection will index kibana_settings.xpack.default_admin_email as null
+			// whereas Metricbeat collection simply won't index it. So if we find kibana_settings.xpack.default_admin_email
+			// is null, we simply remove it
+			if docType == "kibana_settings" {
+				err := legacy.DeleteP("xpack.default_admin_email")
+				if err != nil {
+					return fmt.Errorf("Could not remove default_admin_email field")
+				}
+			}
+
+			return nil
+		}
+	case sm.Product == "logstash":
+		sm.Port = strconv.Itoa(9601)
+
+		sm.handleSpecialCases = func(docType string, legacy *gabs.Container, metricbeat *gabs.Container) error {
+			pipelinesPath := "logstash_stats.pipelines"
+
+			if docType == "logstash_stats" {
+				legacyPipelines := legacy.Path(pipelinesPath)
+				metricbeatPipelines := metricbeat.Path(pipelinesPath)
+
+				legacyPipeline := legacyPipelines.Children()[0]
+				metricbeatPipeline := metricbeatPipelines.Children()[0]
+
+				legacyVertices := legacyPipeline.Path("vertices")
+				metricbeatVertices := metricbeatPipeline.Path("vertices")
+
+				// sort by vertex ID
```

Comment:
Need to deal with sorting this sub-objects

Reply:
@ycombinator I did progress with kibana (tests passing for Kibana and Filebeat). But not sure what to dod here with the vertices. In my execution, MB vertices are null, and they are retrieved as unexpected deletions. Is this correct?